sklearn/tree/_criterion.pyx

```diff
@@ -347,8 +347,9 @@ cdef class ClassificationCriterion(Criterion):
 
         # Reset to pos=start
         self.reset()
+        return 0
```

Comment:
You could do `return self.reset()`, but like Joel points out if `self.reset()` was unsuccessful it returns -1 and skips the execution of subsequent lines (`return 0`)